### PR TITLE
Centralized dummy lookalike generation

### PR DESCRIPTION
**Pull request by TheTrigg**
## About The Pull Request
Man I'm bad at PR names.
Anyways.
Since I used pretty much the exact same code to make dummies that copy people's appearance in my last two PRs, I decided to make a specialized proc for that. It also does it better.
Also squashed a bug while I was at it.

Oh yeah, and another new proc - `/mob/living/carbon/human/copy_clothing_prefs()` - It copies underwear, undershirt and socks because apparently there wasn't a proc that did that before (except `preferences.copy_to()`, but I couldn't rely on the target mob having a client to steal their socks now, could I?)

## Why It's Good For The Game

Less copypastas to achieve the same end result, I guess.

## Changelog
:cl: Trigg
fix: The Select Equipment menu now uses the UI owner's favorite outfits instead of the target mob's. Whoops.
/:cl:

### CODE REVIEW

**Comment by TriggeredBoi** [comment]
In retrospect this PR is way too small and I probably should have snuck these changes in my last one. Too late now, though.

File: code/modules/mob/living/carbon/human/dummy.dm [hunk 1]
```diff
@@ -45,6 +45,30 @@ GLOBAL_LIST_EMPTY(dummy_mob_list)
 	D.in_use = TRUE
 	return D
 
+/proc/generate_dummy_lookalike(slotkey, mob/target)
+	if(!istype(target))
+		return generate_or_wait_for_human_dummy(slotkey)
+
+	var/mob/living/carbon/human/dummy/copycat = generate_or_wait_for_human_dummy(slotkey)
+
+	if(istype(target, /mob/living/carbon))
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/dummy.dm:54]:
```suggestion
	if(iscarbon(target))
```

File: code/modules/mob/living/carbon/human/dummy.dm [hunk 2]
```diff
@@ -45,6 +45,30 @@ GLOBAL_LIST_EMPTY(dummy_mob_list)
...
+			var/mob/living/carbon/human/human_target = target
+			human_target.copy_clothing_prefs(copycat)
+
+		copycat.updateappearance(icon_update=TRUE, mutcolor_update=TRUE, mutations_overlay_update=TRUE)
+	else
+		//even if target isn't a carbon, if they have a client we can make the
+		//dummy look like what their human would look like based on their prefs
+		var/datum/preferences/prefs = target?.client?.prefs
+		if(istype(prefs))
+			prefs.copy_to(copycat, icon_updates=TRUE, roundstart_checks=FALSE, character_setup=TRUE)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/dummy.dm:68]:
```suggestion
		prefs?.copy_to(copycat, icon_updates=TRUE, roundstart_checks=FALSE, character_setup=TRUE)
```

(or even kill the variable and just `target?.client?.prefs?.copy_to(...)`)

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by TriggeredBoi** [comment]
Also, I think a code improvement tag would be fitting; I didn't put anything about code in the changelog because it won't really affect anyone ingame.

**Pull request merged**